src/Tool/ArrayAccessorTrait.php

```diff
@@ -37,7 +37,7 @@ private function getValueByKey(array $data, $key, $default = null)
             $keys = explode('.', $key);
 
             foreach ($keys as $innerKey) {
-                if (!array_key_exists($innerKey, $data)) {
+                if (!is_array($data) || !array_key_exists($innerKey, $data)) {
```

Comment:
The basic test at https://github.com/stevenmaguire/oauth2-client/blob/add-array-accessor-tests/test/src/Tool/ArrayAccessorTraitTest.php#L40 broke it because the dot notation expects two layers and `$data` was set to the string value of the first layer. I don't know how I missed this previously and I am glad I caught it!